### PR TITLE
Npc occupied chair check

### PR DESCRIPTION
**Pull request by jim-huynh**
#### Summary
Bugfixes "Have npcs check that a chair is not occupied before trying to relax on it"

#### Purpose of change
This change is to resolve the problem where two or more npcs had the same chair_pos and push each other out of the way to get the seat until one eventually dies (probably?).

#### Describe the solution
When an NPC starts to move to their relax position (chair_pos), check again that their seat is not occupied.  The occupation check was used originally to find the chair that they first sat on, but someone else could have occupied the chair after they left to do another task.

This check resolves the issue where their chair got stolen.

#### Describe alternatives you've considered
Somehow determine whether the current npc and the sitting npc have the same destination, and if the current npc is in the relaxation mode, then tell this one to find a new chair.
I wasn't able to find out how to determine what mode the current npc is in (relaxation), and the npc would still have to rely on the chair finding code.

#### Testing
Opened up my save that had 6 npcs fighting for the same seat.  After this code change, they no longer fight all day, and quietly occupy different beds.
image

#### Additional context

Bug referenced in: 
https://github.com/CleverRaven/Cataclysm-DDA/issues/51395
https://github.com/CleverRaven/Cataclysm-DDA/issues/47976

### CODE REVIEW

File: src/npcmove.cpp [hunk 1]
```diff
@@ -2570,11 +2570,14 @@ void npc::worker_downtime()
...
     }
-    //  already know of a chair, go there
-    if( chair_pos != tripoint_min ) {
-        if( here.has_flag_furn( ter_furn_flag::TFLAG_CAN_SIT, here.getlocal( chair_pos ) ) ) {
-            update_path( here.getlocal( chair_pos ) );
-            if( pos() == here.getlocal( chair_pos ) || path.empty() ) {
+    //  already know of a chair, go there - if there isn't already another creature there.
+    //  this is a bit of behind the scene omniscience for the npc, since ideally the npc 
+    //  should walk to the chair and then change their destination due to the seat being taken.
+    auto local_chair_pos = here.getlocal( chair_pos );
```
Comment (actual-nh) [inline src/npcmove.cpp:2576]:
```suggestion
    tripoint local_chair_pos = here.getlocal( chair_pos );
```
You will also need to run astyle (with the settings in the main directory `.astylerc` file).

Comment (jim-huynh) [reply src/npcmove.cpp]: > 
> 
> You will also need to run astyle (with the settings in the main directory `.astylerc` file).

Ah, yeah thanks.

**Comment by Maleclypse** [comment]
You have a conflict to resolve.

**Comment by jim-huynh** [comment]
> 
> 
> You have a conflict to resolve.

Thanks for pinging me.  I merged from the upstream and committed the new changes.

**Pull request merged**